### DISCUSSION TITLE
Streamlining Phone Storage with Immich: Auto-Delete After Upload Feature?

### DISCUSSION THREAD

**Discussion by blackangelpr** [immich-app/immich#12576] [General]
Is there an option in the Immich app to automatically delete photos from a device once they've been uploaded? Many users, including families with frequent photo-takers, find it challenging to manage device storage due to the sheer volume of pictures. Having an automatic delete feature would not only free up space but also remove the need for constant manual management. This would allow users to focus more on capturing moments rather than worrying about running out of storage.

Additionally, for frequent travelers, this feature could be particularly beneficial. In certain regions, there might be privacy concerns, and the ability to automatically clear photos from a device after they're safely stored could offer peace of mind.

**Comment by mmomjian** [comment]
Duplicate of #165 , I don’t think either android or iOS allow fully automatic deletion without user input.

**Reply by mkopnsrc** [reply]
It does allow it. Nextcloud App for Android seems to have this feature which auto deletes photos from device once the they are uploaded to Nextcloud server (self-hosted).

**Reply by Mathew-A-Mickle** [reply]
iOS has the ability as well for Nextcloud

**Comment by blackangelpr** [comment]
seems may be could ask after upload for user permission to delete local photos ?  or at least an option to select
https://developer.apple.com/documentation/photokit/phassetchangerequest/1624062-deleteassets

**Reply by mmomjian** [reply]
This is how we currently handle deletions. It opens a pop up to prompt for permission.

**Comment by Hanibachi** [comment]
there doesn't seem to be any updates about the intentions of the devs to add this feature or not.
also ... i m a flutter dev and i m not exactly sure how diffrent is the process for native devs but i don't think there would be any issue with coding up an autoddelete function (from a platform prespective) . free up space is the feature that really put google photos on the map for most people, it would really make immich the deffacto media backup/sync choice compared to something like  photoprism

**Reply by alextran1502** [reply]
As noted in #165, this feature will be implemented after the stable release

**Reply by lionfish2** [reply]
In light of the stable release, does this mean auto-delete after upload is coming soon?

**Comment by rChilLab** [comment]
This feature could be highly valuable for many users wanting to use Immich as a full replacement for iCloud Photos or Google Photos, not just as a backup tool.

The idea:
	•	Automatic upload to Immich.
	•	Optional automatic deletion of local photos after successful upload (similar to how Nextcloud does it using PhotoKit).

This would allow users to fully offload their photos and free up local device storage without relying on cloud services.

Since the last comments on this topic are now around 10 months old, I wanted to kindly ask if there’s any update on the current status of this feature or if it’s still being considered.
This could make Immich a real self-hosted alternative to commercial cloud solutions for many users.

**Reply by alextran1502** [reply]
Yes it will be put in after we reach the stable release

**Reply by ALIFLAPASAR** [reply]
any update regarding this?

**Reply by stephenmoore33** [reply]
A server supporter here..... I switched to Immich.. Spent the last several days uploading my photos off my wife and I's iPhone only to find out that there is no easy way to delete them from iCloud/my device once uploaded..

Really wish this was a feature..

**Comment by JeremyTBradshaw** [comment]
+1 to the votes for this feature.  It's a deal breaker for most anyone who will just be a user and not the Immich admin.

**Comment by tracure1337** [comment]
+ 1 Spam :-)

**Comment by jmakov** [comment]
This still isn't implemented?

**Reply by rChilLab** [reply]
It's implemented with "Free Up Space" feature. https://docs.immich.app/features/mobile-app/#free-up-space

It is not automatically but it works.
